chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -832,7 +856,17 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
         Ok(result)
     }
 
-    pub fn get_block_id_tree_top_as_list(
+    pub fn get_block_id_tree_as_list(&self) -> Result<Vec<Id<Block>>, PropertyQueryError> {
+        self.get_block_ids_by_height(0.into())
+    }
+
+    /// Return ids of all blocks with height bigger than the specified one,
+    /// sorted by height (lower first).
+    // TODO: this function iterates over all block indices in the DB, which is too expensive
+    // for places where it's currently used (such as block invalidation or best chain selection).
+    // We need either to optimize it or replace it with some other solution.
+    // See https://github.com/mintlayer/mintlayer-core/issues/1033, item #5.
+    pub fn get_block_ids_by_height(
```

Comment:
get_block_ids_higher_than()

Reply:
I've decided to use the name `get_higher_block_ids_sorted_by_height` to reflect that the returned blocks both have a height bigger than the specified one and that they are sorted by height.